src/stateMatrixParser.cpp

```diff
@@ -0,0 +1,443 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+
+#include "NormalizedCompressionMarkovTable.h"
+#include "stateMatrixParser.h"
+
+
+
+std::vector<Metrics> mutate_state_matrix(std::string state_matrix_string,unsigned int states, unsigned int  alphabet_size,unsigned int numIt, bool print ){
+    StateMatrix stM = string_to_state_matrix(state_matrix_string,states,alphabet_size);
+    std::vector<unsigned int> kvector{2,3,4,5,6,7,8,9};
+    BestKMarkovTables<NormalizedCompressionMarkovTable> bestMkvTableCompression(kvector, alphabet_size);
+    std::vector<Metrics> m_vector;
+    for (auto index = 0u; index<stM.get_size(); index++){
+        Metrics metrics;
+        bestMkvTableCompression.reset();
+        TuringRecord tr{0,0,0};
+        std::vector<unsigned int> block_nc;
+        auto a = stM.get_element(index);
+
+        stM.v[index] = tr;
+        auto vector_written_string = recreate_tm(stM, states, alphabet_size, numIt);
+        block_nc.insert(block_nc.end(), vector_written_string.begin(), vector_written_string.end());
+        
+        while(tr.next(10,2)){
+            stM.v[index] = tr;
+            auto vector_written_string = recreate_tm(stM, states, alphabet_size, numIt);
+            block_nc.insert(block_nc.end(), vector_written_string.begin(), vector_written_string.end());
+        }
+        
+        //compute best NC
+
+        metrics = bestMkvTableCompression.update_string(block_nc);
+        if(print){
+          std::cout << "--------------------------------------" << std::endl;
+          std::cout << "Rule :" << index +1 << std::endl;
+          std::cout << "best k:" <<metrics.k << std::endl;
+          std::cout << "amplitude:" << metrics.amplitude << std::endl;
+          std::cout << "normalizedCompression:" << metrics.normalizedCompression << std::endl;
+          std::cout << "--------------------------------------" << std::endl;
+        }
+        m_vector.push_back(metrics);
+        stM.v[index]=a;
+    }
+    return m_vector;
+}
+
+std::vector<Metrics> permutate_state_matrix(std::string state_matrix_string,unsigned int states, unsigned int  alphabet_size,unsigned int numIt, bool print ){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'state_matrix_string' should be passed by const reference.](https://app.codacy.com/app/jorgeMFS/TMCompression/pullRequest?prid=4494377)